### PR TITLE
DOC: Improve docs (GH19312) for Series.nonzero()

### LINKED ISSUE #19312: Series.nonzero(): Returns locations, not indices
#### Problem description
Documentation says

Series.nonzero(): Return the indices of the elements that are non-zero

while it actually seems to return the integer locations.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: AMD64 Family 21 Model 16 Stepping 1, AuthenticAMD
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.21.0
pytest: None
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: None
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.1.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by drorata**
- [x] closes #19312 
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -490,13 +490,14 @@ def compress(self, condition, *args, **kwargs):
 
     def nonzero(self):
         """
-        Return the indices of the elements that are non-zero
+        Return the integer indices of the elements that are non-zero
```
Comment (jreback) [inline pandas/core/series.py]: we normally call these indexers

File: pandas/core/series.py [hunk 2]
```diff
@@ -490,13 +490,14 @@ def compress(self, condition, *args, **kwargs):
...
 
         This method is equivalent to calling `numpy.nonzero` on the
         series data. For compatibility with NumPy, the return value is
         the same (a tuple with an array of indices for each dimension),
         but it will always be a one-item tuple because series only have
-        one dimension.
+        one dimension. Note that the method returns the *integer* indices
```
Comment (jreback) [inline pandas/core/series.py]: this is just repeating from above yes?

Comment (jreback) [reply pandas/core/series.py]: better to add this to a Notes section (and you could move most of the above text there as well)

File: pandas/core/series.py [hunk 3]
```diff
@@ -508,6 +509,15 @@ def nonzero(self):
         3    4
         dtype: int64
 
+        >>> s = pd.Series([0, 3, 0, 4])
+        >>> s.index=['a', 'b', 'c', 'd']
+        >>> s.nonzero() # same return although index of s is different
```
Comment (jreback) [inline pandas/core/series.py]: you can put the comment before the example

File: pandas/core/series.py [hunk 4]
```diff
@@ -508,6 +509,15 @@ def nonzero(self):
         3    4
         dtype: int64
 
+        >>> s = pd.Series([0, 3, 0, 4])
```
Comment (jreback) [inline pandas/core/series.py]: just directly construct the Series with the index, then show it

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (drorata) [reply pandas/core/series.py]: The confusion reflected in #19312 relates to the fact that the definitions are not clear. Personally, I don't know what are the different definitions. The `index` of a `Series` may be a default one; namely integers starting from `0`. Or, named indices. This PR stress in the documentation of `nonzero` that it returns the integers.

File: pandas/core/series.py (hunk 2 above)
Comment (drorata) [reply pandas/core/series.py]: I removed the addition and only stressed the properties of the returned value in the first one-liner. I guess some glossary could be helpful to clarify what are the differences between the different definitions.

**Comment by drorata** [comment]
@jreback I have updated the PR

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @drorata

**Pull request merged**